### ISSUE TITLE
bug in ni_Htest_sortGTI: discrepancy between grid search results and fixed energy range

### ISSUE THREAD

**Issue by sguillot** [paulray/NICERsoft#67]
I noticed another unexpected behaviour from ni_Htest_sortGTI.py.
When using the results from a grid search (i.e., optimal energy range) into a single run, the output significance, exposure time and Nb of events are different than what the grid search gives.

For example, with the following execution:
`ni_Htest_sortgti.py @TEST.list newtest --gridsearch --emin=0.27 --maxemin=0.3 --minemax=1.22 --emax=1.26`
gives as a result:
```
Maximum significance: 6.847 sigma
   obtained in 414.76 (out of 414.82 ksec)
   between 0.29 and 1.24 keV
   for 149230 events
```

Then, if I use this optimal energy range (0.29-1.24 keV), with the following command:
`ni_Htest_sortgti.py @TEST.list newtest2 --emin=0.29 --emax=1.24`
the result is:
```
Maximum significance: 6.628 sigma
   obtained in 406.95 ksec (out of 414.82 ksec)
   for 144175 events
```

But the significance, exposure time and Nb of events should be the same at the grid search run above.  Note that this happens with different data sets (different pulsars).

I have found yet why this is different, and I'll investigate in the next few days.
@kerrm, if you spot something obvious, please let me know.

**Comment by kerrm** [comment]
Recently I found and fixed an off-by-one bug, where the gti selection was
not correct.  The code was determining "amax" and selecting gti[:amax],
when it should have been gti[:amax+1].  If it's related to this bug, it's
possible I missed some other place it should have been fixed between the
"dosearch" part of the code and the other one.

If that's not it, I'll take a closer look at it soon.

**Comment by kerrm** [comment]
PS -- perhaps we can add a simple test at the end of the script to help
catch future inconsistencies, e.g. by insisting that the H (or Z) test on
the resulting event selection gives the same as was determined during the
optimization!

**Comment by sguillot** [comment]
This has been fixed in #68
